### PR TITLE
Matter Switch: improve child device profile support

### PR DESCRIPTION
**Pull request by ctowns**
Child device profiles will now be determined dynamically based on the device type of the child device endpoint. This will improve support for multi-devices.

### CODE REVIEW

File: drivers/SmartThings/matter-switch/src/init.lua [hunk 1]
```diff
@@ -56,6 +70,26 @@ local function find_default_endpoint(device, component)
...
+    if ep.endpoint_id == child_ep then
+      -- Some devices report multiple device types which are a subset of
+      -- a superset device type (For example, Dimmable Light is a superset of
+      -- On/Off light). This mostly applies to the four light types, so we will want
+      -- to match the profile for the superset device type. This can be done by
+      -- matching to the device type with the highest ID
+      local id = 0
+      for _, dt in ipairs(ep.device_types) do
+        id = math.max(id, dt.device_type_id)
+        profile = device_type_profile_map[id]
```
Comment (greens) [inline drivers/SmartThings/matter-switch/src/init.lua:85]: quibble, but you could do this outside the loop, right?

Comment (ctowns) [reply drivers/SmartThings/matter-switch/src/init.lua]: Good point, moved the assignment outside the loop 👍

**Review by greens — APPROVED** [review]

**Comment by ctowns** [comment]
Moved profile assignment out of device type loop and added additional unit tests for Extended Color Light child device

File: drivers/SmartThings/matter-switch/src/test/test_multi_switch_parent_child_lights.lua [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+				mock_device.id,
+				clusters.LevelControl.server.attributes.CurrentLevel:build_test_report_data(mock_device, child1_ep, 50)
+			}
+		},
+		{
+			channel = "capability",
+			direction = "send",
+			message = mock_children[child1_ep]:generate_test_message("main", capabilities.switchLevel.level(math.floor((50 / 254.0 * 100) + 0.5)))
+		},
+	}
```
Comment (greens) [inline drivers/SmartThings/matter-switch/src/test/test_multi_switch_parent_child_lights.lua:256]: nit: tabs here

**Review by greens — APPROVED** [review]

**Comment by ctowns** [comment]
This was tested with Hue Dymera and Aqara LED Light Strip as indicated in [CHAD-12231](https://smartthings.atlassian.net/browse/CHAD-12231).

Also, unit tests have been added for checking both multi-plug and multi-light devices.

**Pull request merged**